### ISSUE TITLE
Main targets are compiled but others fail to recompile

### ISSUE THREAD

**Issue by arctouch-rafaellucena** [johnno1962/InjectionIII#292]
I built a framework that is linked to a separate target from the Main target in Xcode.
The recompilation fails for the framework target.

```
💉 *** Re-compilation failed (see: /var/folders/r7/sph9bhc55hq988t4fyy0vh6w0000gn/T/com.johnholdsworth.InjectionIII//command.sh)
/Users/rafaellucena/Library/Developer/Xcode/DerivedData/ProjectName-aoioyfrosgutrfbhjtxbdhwqnuro/Build/Intermediates.noindex/ProjectName.build/Internal_Debug-iphonesimulator/FrameworkName.build/module.modulemap:2:19: error: umbrella header 'FrameworkName.h' not found
  umbrella header "FrameworkName.h"
                  ^
<unknown>:0: error: could not build Objective-C module 'FrameworkName'
```

Any thoughts?

ps: `ProjectName` refers to the project name and `FrameworkName` refers to the framework name.

**Comment by johnno1962** [comment]
Hi, this is probably because of the directory the compiler is invoked in which for injection is always the main source root. The simplest way to resolve this is to modify the "Header Search Path" of the project to include the framework target root.

**Comment by arctouch-rafaellucena** [comment]
Maybe I'm doing something wrong but here is how things are right now:

My project hierarchy:
```
- ProjetctName.xcodeproj
- ProjetctName.xcworkspace
- FrameworkName
    - FramworkName.h file 
```

What I've tried with no success:
1. Update `Header Search Paths` to include: `$(PROJECT_DIR)/FrameworkName`
2. Update `Framework Search Paths` to include `$(PROJECT_DIR)/FrameworkName`
3. Update `User Header Search Paths ` to include `$(PROJECT_DIR)/FrameworkName`
4. Also tried the combination of the three options above

Following how the frameworks headers were already linked into `Header Search Paths` I also tried to use the same pattern: `build/internal_Debug-iphoneos/FrameworkName/FrameworkName.framework/Headers`.

Or maybe I completely misunderstood what you suggested.
Any thoughts?

**Comment by johnno1962** [comment]
No you're following me but .modulemap files are renowned for their inability to follow include paths. The first step to see if we can find a fix is to look at the 'command.sh' file the message mentions and see if you can get it to work. If it works only if you execute it in the $(PROJECT_DIR)/FrameworkName directory, I can make a small change to injection to make sure the recompilation command is executed in that directory. You seem to be injecting a file in the framework. Would that be true?

**Comment by johnno1962** [comment]
I've been unable to replicate this problem this end alas. Is there any chance you might be able to post a small example project displaying the problem for me to investigate?

**Comment by rlmg-eero** [comment]
Hey @johnno1962 sorry for the delay.
I had a company transition.

So let me answer your question:
The framework is a dependency for the main project.

I don't know why it is trying to rebuild it since there is no change related 🤔 (I'm trying to inject a view controller that has nothing to do with the framework - not even close).
Maybe I don't have enough expertise on this and might be thinking away from the problem solution.

I tried checking the `command.sh` file (as you suggested) but couldn't find any reference to the framework there 🤔.
I tried to manually edit (based on the pattern I could see for other frameworks) but this file seems to be autogenerated (that means my changes were never applied since it is regenerated).

The framework has its own `Podfile` section so it can install its dependencies.
Basically I have a target on `Podfile` explicitly for the framework's dependencies.
Does that impact on anything?

**Comment by rlmg-eero** [comment]
> I've been unable to replicate this problem this end alas. Is there any chance you might be able to post a small example project displaying the problem for me to investigate?

I tried but no success in reproducing the issue.
I even tried to add the same Podfile into the sample project and it works like charm.
Considering this, I'd say it is a project specific problem. Maybe a "wrong" configuration on `Project.xcproj` or `Project.xcworkspace`.

Does it ring a bell?

**Comment by johnno1962** [comment]
Thanks for getting back. Sounds like it's a confused DerivedData type of issue. If you're interested, I'm breaking in a new version of Injection as a Swift Package you might want to try: https://github.com/johnno1962/HotReloading. It's all the same code but a lot easier to develop/debug and work on in general than having a separate app.

**Comment by rlmg-eero** [comment]
I loved the idea of this being an external tool so I could avoid adding more dependencies to the project.
I just tried `https://github.com/johnno1962/HotReloading` and it works like charm 😍.
I'll try a little bit more using the external tool solution. If I don't succeed then I'll go through the dependencies checking process to be approved to use it.
@johnno1962 thanks a lot for your time and quick responses. I'll get back to you if I figure out what is wrong with my project.

**Comment by rlmg-eero** [comment]
Now I have a different problem haha.
I have two options available:
1. Use SPM HotReloading library (https://github.com/johnno1962/HotReloading)
2. Try to figure out the problem using InjectionIII as an external tool

The limitation for the first option is: I don't want the library (that is added to the target) to be in my release builds. Basically, I want to avoid shipping in the bundle for the customer. Do you have any thoughts? Do you know any way of modifying the `xcodeproj` file before build time and leave it as it is configured after?

The limitation for the second option (that would be the best): I couldn't figure out the problem we were discussing above hahah. I'd guess investing more time in fixing this second option is a better alternative.

But I'm open to new suggestions if you have hahaha.

**Comment by johnno1962** [comment]
It's an interesting dilemma. If you still have the problem we discussed, I'm finding the new HotReloading incarnation much easier to work with but it's a bit of a nuisance having to add it and then having to remember to remove it before releasing. There doesn't seem to be a way to have conditional inclusion of packages by configuration so I could make all the code #if DEBUG but I'm not sure the code would survive a formal review. For now, sounds like you're using the package to track down problems, then file a PR the external tool as it is the same code base.

**Comment by rlmg-eero** [comment]
I'm searching for ways to include the library for specific schemes and then removing it after the IPA is generated.
Not even sure it is possible but might be a way to make it work using HotReloading.

Meanwhile, I'm still trying to figure out the problem we discussed.
I'll get back to you if I have more ideas on how to solve it. 
Maybe it can be a problem someone is facing as well.

**Comment by johnno1962** [comment]
I'm confused, you still have the header files problem? I thought it went away or does it crop up from time to time? I've consolidated the HotReloading and InjectionIII code to be the same so if you want to explore the problem, use the package version and drag a clone on disk into your project to override in real time as it mentions in the README.
